### PR TITLE
Make user and client keys endpoints pass Pedant specs

### PR DESCRIPTION
**Pull request by jrunning**
Depends on chef/chef#4587 and chef/chef-server#740.

- Implement ActorKeyEndpoint, ActorDefaultKeyEndpoint, ActorKeysEndpoint.
- Implement user, client keys in `ActorEndpoint#delete`, `#put`.

- RestBase
  - Fix RestErrorResponse exceptions to report actual `rest_path`
    instead associated with the failed data store operation instead of
    `request.rest_path`.
  - Move `json_response`, `already_json_response` args `request_version`
    and `response_version` into options hash; add docs.

- DataError, RestErrorResponse: Pass useful message text to `super`.

- RestRouter: Clean up logging
  - Print request methods, paths and bodies more readably for log_level >= INFO.
  - Pretty-print RestRequest objects (only printed when log_level == DEBUG).

- Server: Change default log_level to `:warn` (to enable logging cleanup above).

- `Rakefile`, `spec/run_oc_pedant.rb`
  - Consume RSpec, Pedant options from `ENV['RSPEC_OPTS']`, `ENV['PEDANT_OPTS']` (see `rake -D`).
  - Consume `ENV['LOG_LEVEL'` (see `rake -D`).
  - Clean up ChefZero::Server default opts and move duplicated logic to `start_chef_server` method.

### CODE REVIEW

File: lib/chef_zero/endpoints/actor_key_endpoint.rb [hunk 1]
```diff
@@ -0,0 +1,97 @@
...
+        delete_data(request, key_path)
+
+        already_json_response(200, data)
+      end
+
+      def put(request)
+        # We grab the old data to trigger a 404 if it doesn't exist
+        get_data(request, data_path(request))
+
+        set_data(request, path, request.body)
```
Comment (danielsdeleo) [inline lib/chef_zero/endpoints/actor_key_endpoint.rb]: Seems odd that this does not need a special case for the default key

File: spec/run_oc_pedant.rb [hunk 2]
```diff
@@ -104,7 +146,8 @@ def start_cheffs_server(chef_repo_path)
     # are turned off" - @jkeiser
     #
     # ...but we're not there yet
-    '--skip-keys',
+    '--skip-controls',
+    '--skip-acl',
```
Comment (danielsdeleo) [inline spec/run_oc_pedant.rb]: Why did these get added?

**Comment by danielsdeleo** [comment]
There were two things I didn't understand. Also, I'm assuming we need to merge the chef change to make travis happy. Everything else is :+1: Really like the logging and test running improvements.

File: lib/chef_zero/endpoints/actor_key_endpoint.rb (hunk 1 above)
Comment (jrunning) [inline lib/chef_zero/endpoints/actor_key_endpoint.rb]: Hmm, that does seem odd. I'll take a closer look.

File: spec/run_oc_pedant.rb (hunk 2 above)
Comment (jrunning) [inline spec/run_oc_pedant.rb]: I'm not sure. Maybe a bad rebase, or maybe I added them temporarily and forgot to take them out. I'll do that now.

**Comment by danielsdeleo** [comment]
@chef/client-dev-tools can I get another review on this?

**Comment by jkeiser** [comment]
:+1:, this has a lot of great simplification too! Awesome job.

**Pull request merged**